Mathlib/Topology/Separation/Basic.lean

```diff
@@ -834,6 +834,15 @@ theorem isClosedEmbedding_update {ι : Type*} {β : ι → Type*}
   apply isClosed_set_pi
   simp [forall_update_iff, hs]
 
+lemma Function.update_eventuallyEq_nhdsNE
+    {α β : Type*} [TopologicalSpace α] [T1Space α] [DecidableEq α] (f : α → β) (a a' : α) (b : β) :
+    Function.update f a b =ᶠ[𝓝[≠] a'] f := by
```

Comment:
I agree that the `nhdsNE` version is likely the most useful in practice, but can you also provide the `principal_singleton_compl` version as the most general, which people can use with mono to get the result for any other filter they may need?

Reply:
Are you suggesting to prove `update f a b =ᶠ[𝓟 {a'}ᶜ] f`? 

I think that isn't true, unfortunately. In the proof I wrote, the filter is `𝓟 {a'}ᶜ ⊓ 𝓝 a'`, but one branch of the `rcases` uses properties of `𝓟 {a'}ᶜ` and the other uses properties of `𝓝 a'`. The properties of `𝓝 a'` that are required are non-vacuous, and it is here that the `T1Space` assumption is needed; we're requiring that `𝓝 a'` be disjoint from `pure a` for any `a ≠ a'` (i.e. the conclusion of `disjoint_nhds_pure`).

Reply:
PS. I thought a bit about this and found a better generalization: prove the function-update result for any filter extending `cofinite`, and then show that `nhdsNE` extends `cofinite` for T1 spaces. What do you think?